tests/_internal/mbed_tools/test_config_command.py

```diff
@@ -0,0 +1,121 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+import json
+import pathlib
+from unittest import TestCase, mock
+
+from click.testing import CliRunner
+from tabulate import tabulate
+
+from mbed_build._internal.mbed_tools.config import (
+    config,
+    _build_tabular_output,
+    _build_json_output,
+    _build_legacy_output,
+)
+from tests._internal.config.factories import BuildConfigFactory, OptionFactory
+
+
+@mock.patch("mbed_build._internal.mbed_tools.config.assemble_config", return_value=BuildConfigFactory())
+class TestConfig(TestCase):
+    mbed_target = "K64F"
+    project_path = "somewhere"
+
+    def test_config_table(self, assemble_config):
+        runner = CliRunner()
+        result = runner.invoke(config, ["-m", self.mbed_target, "-p", self.project_path])
+
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(result.output, _build_tabular_output(assemble_config.return_value) + "\n")
```

Comment:
Ah.. the infamous trailing new line. Have you considered other options for asserting?
1) `self.assertIn(expected_output, result.output)` - should be ok, as we assert that a big, non repeatable chunk of text exists in the output
2) `self.assertEqual(result.output.strip(), expected_output)` - potentially problematic if `_build_tabular_output` has whitespace at either beginning or end

I think `assertIn` would be most elegant here. Not a major thing though. ;)

Reply:
8806187